File: ShiftSync.Web/src/types/shiftsync/shift.ts
export interface Shift {
  id: number;
  name: string;
  startTime: string; // "HH:mm:ss"
  endTime: string;   // "HH:mm:ss"
  maxAllowedBreaksDurationMinutes: number;
  minActiveEmployeesRequired: number;
  isActive: boolean;
}

export interface CreateShiftDto {
  name: string;
  startTime: string;
  endTime: string;
  maxAllowedBreaksDurationMinutes: number;
  minActiveEmployeesRequired: number;
}

export interface UpdateShiftDto {
  name?: string;
  startTime?: string;
  endTime?: string;
  maxAllowedBreaksDurationMinutes?: number;
  minActiveEmployeesRequired?: number;
}

export interface ShiftFilter {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  isActive?: boolean;
}

File: ShiftSync.API/Models/Entities/Shift.cs
namespace ShiftSync.API.Models.Entities;

public class Shift
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public TimeSpan StartTime { get; set; }
    public TimeSpan EndTime { get; set; }

    public int MaxAllowedBreaksDurationMinutes { get; set; } = 60;

    public int MinActiveEmployeesRequired { get; set; } = 2;

    public bool IsActive { get; set; } = true;

    public ICollection<UserShift> UserShifts { get; set; } = new List<UserShift>();
}
